test/MassMigration.test.ts

```diff
@@ -150,4 +130,100 @@ describe("Mass Migrations", async function() {
             "Good state transition should not rollback"
         );
     });
+    it("submit a batch, finalize, and withdraw", async function() {
+        const { rollup, withdrawManager, testToken, vault } = contracts;
+        const tx = new TxMassMigration(
+            Alice.stateID,
+            USDT.castInt(39.99),
+            1,
+            USDT.castInt(0.01),
+            Alice.nonce + 1,
+            USDT
+        );
+        const leaf = State.new(
+            Alice.pubkeyIndex,
+            tokenID,
+            tx.amount,
+            0
+        ).toStateLeaf();
+        const withdrawTree = Tree.merklize([leaf]);
+        const { safe } = stateTree.applyMassMigrationBatch([tx]);
+        assert.isTrue(safe);
+
+        const commitment = MassMigrationCommitment.new(
+            stateTree.root,
+            await registry.root(),
+            mcl.aggreagate([Alice.sign(tx)]),
+            tx.spokeID,
+            withdrawTree.root,
+            tokenID,
+            tx.amount,
+            tx.encode()
+        );
+
+        const batch = commitment.toBatch();
+        await batch.submit(rollup, TESTING_PARAMS.STAKE_AMOUNT);
+
+        const batchId = Number(await rollup.numOfBatchesSubmitted()) - 1;
+
+        await expect(
+            withdrawManager.ProcessWithdrawCommitment(batchId, batch.proof(0))
+        ).revertedWith("Vault: Batch shoould be finalised");
+
+        await mineBlocks(ethers.provider, TESTING_PARAMS.TIME_TO_FINALISE);
+
+        // We cheat here a little bit by sending token to the vault manually.
+        // Ideally the tokens of the vault should come from the deposits
+        await testToken.transfer(vault.address, tx.amount);
```

Comment:
Haven't had a way to finalise deposit well with current utils, so doing a manual transfer here.